src/pages/signup/SignupForm.tsx

```diff
@@ -0,0 +1,270 @@
+import { Field, Formik, FormikActions } from 'formik';
+import * as React from 'react';
+import { Button, Col, FormFeedback, FormText, Form, FormGroup, Input, Label, Row } from 'reactstrap';
+import { compose, graphql } from 'react-apollo';
+import * as Yup from 'yup';
+
+import { CreateUser, CREATE_OR_LOGIN_WITH_PROVIDERS, CREATE_USER, User } from 'networking/users';
+import { auth, login, signInWithGooglePopUp } from 'utils/firebase';
+import { getFriendlyErrorMessage } from 'utils/validators';
+
+interface SignupProps {
+  createOrLoginWithProviders: (id: string) => Promise<any>;
+  createUser: (user: CreateUser) => Promise<User>;
+}
+
+const LoginSchema = Yup.object().shape({
+  signupEmail: Yup.string()
+    .email('Please enter a valid email address.')
+    .required('Please enter an email address.'),
+  signupFirstName: Yup.string()
+    .required('Please enter your first name.'),
+  signupLastName: Yup.string()
+    .required('Please enter your last name.'),
+  signupPassword: Yup.string()
+    .min(8, 'Your password is too short, please enter a password that is at least 8 characters long')
+    .required('Please enter a valid password.'),
+  signupConfirmPassword: Yup.string()
+    .oneOf([Yup.ref('signupPassword')], 'Your passwords do not match.')
+    .required('Please re-enter your password here.'),
+});
+
+interface SignupFormInput {
+  signupEmail: string;
+  signupFirstName: string;
+  signupLastName: string;
+  signupPassword: string;
+  signupConfirmPassword: string;
+  authenticationError: string;
+}
+
+interface State {
+  providerError: string | null;
+}
+
+class SignupForm extends React.Component<SignupProps, State> {
+  readonly state = {
+    providerError: null,
+  };
+
+  render() {
+    return (
+      <Formik
+        initialValues={{
+          signupEmail: '',
+          signupFirstName: '',
+          signupLastName: '',
+          signupPassword: '',
+          signupConfirmPassword: '',
+          authenticationError: '',
+        }}
+        validationSchema={LoginSchema}
+        onSubmit={this.handleSubmit}
+      >
+        {({ errors, touched, setFieldValue, submitForm, isSubmitting }) => (
+          <Form method="POST">
+            <div className="mb-7">
+              <h2 className="h3 text-primary font-weight-normal mb-0">
+                Welcome to <span className="font-weight-semi-bold">Beenest</span>
+              </h2>
+              <p>Fill out the form to get started.</p>
+            </div>
+
+            <Row className="d-flex flex-sm-column flex-md-row">
```

Comment:
i was thinking about moving these details to a helper array of objects and map it out to avoid boilerplate bloat

Reply:
actually might be too many variables to cover to cleanly do this at this moment